### PR TITLE
Export name and typeDescription members of reflector struct

### PR DESCRIPTION
**Pull request by alexanderstephan**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Users can provide their own error handler by setting the `DefaultWatchErrorHandler` in the `cache.Options`. This function takes the reflector (`toolscache.Reflector`) as input. However, although this option for extensibility is provided, most members of the `Reflector` are not exported. I would argue that at least the `Name` and the `TypeDescription` are needed. Otherwise, not even the default implementation
```go
// DefaultWatchErrorHandler is the default implementation of WatchErrorHandler
func DefaultWatchErrorHandler(r *Reflector, err error) {
	switch {
	case isExpiredError(err):
		// Don't set LastSyncResourceVersionUnavailable - LIST call with ResourceVersion=RV already
		// has a semantic that it returns data at least as fresh as provided RV.
		// So first try to LIST with setting RV to resource version of last observed object.
		klog.V(4).Infof("%s: watch of %v closed with: %v", r.Name, r.TypeDescription, err)
	case err == io.EOF:
		// watch closed normally
	case err == io.ErrUnexpectedEOF:
		klog.V(1).Infof("%s: Watch for %v closed with unexpected EOF: %v", r.Name, r.TypeDescription, err)
	default:
		utilruntime.HandleError(fmt.Errorf("%s: Failed to watch %v: %v", r.Name, r.TypeDescription, err))
	}
}
```
can be implemented.

#### Which issue(s) this PR fixes:

No issue exists yet.

#### Special notes for your reviewer:
1. Please double-check whether I found all occurrences. In most cases, name actually refers to the concept of a name, and not the variable name as far as I could tell. 

2. I don't see these to members being critical. Alternatively, getters could be implemented.

#### Does this PR introduce a user-facing change?

```release-note
The members name and typeDescription of the Reflector struct are now exported to allow for better user extensibility.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
Left blank for now. I don't see any use for documentation here as the change is so minor.

### CODE REVIEW

**Review by cici37 — COMMENTED** [review]
The main concern in this PR would be in what level the backward compatibility required in this case and if we wanna expand the surface of exposure.

Exporting name and typeDescription sounds reasonable to me. /cc main client maintainer @yliaog to see if there is any concern. :)

File: staging/src/k8s.io/client-go/tools/cache/reflector.go [hunk 1]
```diff
@@ -149,13 +149,13 @@ func DefaultWatchErrorHandler(r *Reflector, err error) {
...
-		klog.V(4).Infof("%s: watch of %v closed with: %v", r.name, r.typeDescription, err)
+		klog.V(4).Infof("%s: watch of %v closed with: %v", r.Name, r.TypeDescription, err)
 	case err == io.EOF:
 		// watch closed normally
 	case err == io.ErrUnexpectedEOF:
-		klog.V(1).Infof("%s: Watch for %v closed with unexpected EOF: %v", r.name, r.typeDescription, err)
+		klog.V(1).Infof("%s: Watch for %v closed with unexpected EOF: %v", r.Name, r.TypeDescription, err)
 	default:
-		utilruntime.HandleError(fmt.Errorf("%s: Failed to watch %v: %v", r.name, r.typeDescription, err))
+		utilruntime.HandleError(fmt.Errorf("%s: Failed to watch %v: %v", r.Name, r.TypeDescription, err))
```
[the remark this answers is not in the document]
Comment (alexanderstephan) [reply staging/src/k8s.io/client-go/tools/cache/reflector.go]: Thanks, fixed.

**Comment by alexanderstephan** [comment]
Hi @yliaog, I just wanted to ask again what your opinion on this change is. Thanks!

**Comment by yliaog** [comment]
is it better to use a function, say Name() { return name} ? you only need to get the name, not need to set it, if i understand it correctly.

**Comment by alexanderstephan** [comment]
Yes, actually I thought this would be more flexible, but I cannot really see a use case where you would want to change the name.

**Comment by alexanderstephan** [comment]
@yliaog Added the getters now.

**Comment by yliaog** [comment]
thanks for the PR

/lgtm
/approve

**Comment by yliaog** [comment]
it's simpler, and better to just use Name() instead of GetName(), @alexanderstephan do you mind to make that change?

**Comment by alexanderstephan** [comment]
I didn't have the time yet, I will do it sometime this weekend.

**Comment by alexanderstephan** [comment]
@yliaog Here is the PR: https://github.com/kubernetes/kubernetes/pull/128035

**Pull request merged**